### PR TITLE
Add auth for Profile and Logout in the menu

### PR DESCRIPTION
**Pull request by sewagodimo**

### CODE REVIEW

File: nurseconnect/templates/core/tags/section_listing_menu.html [hunk 1]
```diff
@@ -25,16 +25,23 @@ <h1 class="HeroTout-headline">
             {% trans "Search" %}
         </a>
       </li>
+      {% if request.user.is_authenticated %}
       <li class="Menu-navItem" role="presentation">
         <a href="{% url 'molo.profiles:view_my_profile' %}" class="Menu-navLink" role="menuitem">
           {% trans "Your Profile" %}
         </a>
       </li>
+
```
Comment (sewagodimo) [inline nurseconnect/templates/core/tags/section_listing_menu.html]:
```suggestion
```

File: nurseconnect/tests/test_views.py [hunk 2]
```diff
@@ -18,13 +18,18 @@ class MenuTestCase(MoloTestCaseMixin, TestCase):
...
         self.client = Client()
+        self.section = self.mk_section(self.section_index,
+                                       title="Section")
 
     def test_menu_renders_correctly(self):
         # Content in this view is dependent on there being sections.
         # So we'll test that the view returns successfully and leave
         # it at that
         response = self.client.get(reverse("menu"))
+
```
Comment (sewagodimo) [inline nurseconnect/tests/test_views.py]:
```suggestion
```

File: nurseconnect/tests/test_views.py [hunk 3]
```diff
@@ -18,13 +18,17 @@ class MenuTestCase(MoloTestCaseMixin, TestCase):
...
+                                       title="Section")
 
     def test_menu_renders_correctly(self):
         # Content in this view is dependent on there being sections.
         # So we'll test that the view returns successfully and leave
         # it at that
         response = self.client.get(reverse("menu"))
         self.assertEqual(response.status_code, 200)
+        self.assertTemplateUsed(response,
+                                "core/tags/section_listing_menu.html")
```
Comment (moh-moola) [inline nurseconnect/tests/test_views.py:42]: You might want to assert the two cases, logged in and logged out.
assert for something obvious like "Login in response" and  "Login not in response"

Comment (sewagodimo) [reply nurseconnect/tests/test_views.py:42]: @moh-moola I have added them

**Review by moh-moola — APPROVED** [review]

File: nurseconnect/templates/core/tags/section_listing_menu.html [hunk 4]
```diff
@@ -35,6 +37,10 @@ <h1 class="HeroTout-headline">
           {% trans "Log out" %}
         </a>
       </li>
+      {% else %}
+      <li class="Menu-navItem" role="presentation">
+        <a href="{% url "auth_login" %}?next={{request.path}}" class="Menu-navLink" role="menuitem">{% trans "Login" %}</a>
+      </li>
+      {% endif %}
     </ul>
```
Comment (codiebeulaine) [inline nurseconnect/templates/core/tags/section_listing_menu.html:45]: shouldn't the ul and nav be inside the if

File: nurseconnect/tests/test_views.py [hunk 5]
```diff
@@ -25,6 +25,20 @@ def test_menu_renders_correctly(self):
...
+        self.assertNotContains(response, "Profile")
+        self.assertContains(response, "Login")
+
+    def test_menu__user_authenticated(self):
```
Comment (codiebeulaine) [inline nurseconnect/tests/test_views.py:33]: is there a test for when the user is not authenticated?

File: nurseconnect/templates/core/tags/section_listing_menu.html (hunk 4 above)
Comment (sewagodimo) [reply nurseconnect/templates/core/tags/section_listing_menu.html:45]: @codiebeulaine  no that ul is declared before the if

**Pull request merged**